modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgnitePdsDiscoDataHandlingInNewClusterTest.java

```diff
@@ -0,0 +1,195 @@
+                    .setDefaultDataRegionConfiguration(
+                        new DataRegionConfiguration()
+                            .setPersistenceEnabled(true)
+                    )
+            );
+
+            cfg.setCacheConfiguration(
+                new CacheConfiguration(STATIC_CACHE_NAME_0)
+                    .setGroupName(MIXED_CACHES_GROUP_NAME_0)
+                    .setAffinity(AFFINITY)
+                    .setNodeFilter(nodeFilter)
+            );
+        }
+
+        if (igniteInstanceName.equals(NODE_CONS_ID_0)) {
+            failingOnNodeJoinSpi.setIpFinder(sharedStaticIpFinder);
+            failingOnNodeJoinSpi.setJoinTimeout(60_000);
+
+            cfg.setDiscoverySpi(failingOnNodeJoinSpi);
+        }
+
+        return cfg;
+    }
+
+    /**
+     * @throws Exception
+     */
+    @Test
+    @WithSystemProperty(key = IgniteSystemProperties.IGNITE_DUMP_THREADS_ON_FAILURE, value = "false")
+    public void testNewDynamicCacheDoesntStartOnOldNode() throws Exception {
+        IgniteEx ig0 = startGrid(NODE_CONS_ID_0);
+
+        startGrid(NODE_CONS_ID_1);
+
+        ig0.cluster().active(true);
+
+        startDynamicCache(ig0, DYNAMIC_CACHE_NAME_0, MIXED_CACHES_GROUP_NAME_0);
+
+        stopGrid(NODE_CONS_ID_1);
+
+        startDynamicCache(ig0, DYNAMIC_CACHE_NAME_1, MIXED_CACHES_GROUP_NAME_0);
+
+        startDynamicCache(ig0, DYNAMIC_CACHE_NAME_2, DYNAMIC_CACHES_GROUP_NAME_1);
+
+        SHOULD_FAIL.set(true);
+
+        IgniteEx ig1 = startGrid(NODE_CONS_ID_1);
+
+        verifyCachesAndGroups(ig1);
+    }
+
+    /** */
+    private void startDynamicCache(Ignite ig, String cacheName, String groupName) {
+        ig.getOrCreateCache(new CacheConfiguration<>(cacheName)
+            .setGroupName(groupName)
+            .setAffinity(new RendezvousAffinityFunction(false, 16))
+            .setNodeFilter(nodeFilter)
+        );
+    }
+
+    /** */
+    private void verifyCachesAndGroups(IgniteEx ig) {
+        Map<String, DynamicCacheDescriptor> caches = ig.context().cache().cacheDescriptors();
+
+        assertEquals(3, caches.size());
+        assertTrue(caches.keySet().contains(GridCacheUtils.UTILITY_CACHE_NAME));
+        assertTrue(caches.keySet().contains(STATIC_CACHE_NAME_0));
+        assertTrue(caches.keySet().contains(DYNAMIC_CACHE_NAME_0));
+
+        Map<Integer, CacheGroupDescriptor> groups = ig.context().cache().cacheGroupDescriptors();
+
+        assertEquals(2, groups.size());
+
+        boolean defaultGroupFound = false;
+        boolean mixedCachesGroupFound = false;
+
+        for (CacheGroupDescriptor grpDesc : groups.values()) {
+            if (grpDesc.cacheOrGroupName().equals(GridCacheUtils.UTILITY_CACHE_NAME))
+                defaultGroupFound = true;
+            else if (grpDesc.cacheOrGroupName().equals(MIXED_CACHES_GROUP_NAME_0))
+                mixedCachesGroupFound = true;
+        }
+
+        assertTrue(defaultGroupFound && mixedCachesGroupFound);
```

Comment:
Assert message can be re-used from non-persistent test.

Reply:
Fixed.